### DISCUSSION TITLE
creating-a-source-generator-part-8-solving-the-source-generator-marker-attribute-problem-part2/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#367] [Announcements]
# creating-a-source-generator-part-8-solving-the-source-generator-marker-attribute-problem-part2/

In this post I describe how I solved a problem I've been wrestling with around source generators: where to put the 'marker attributes'

https://andrewlock.net/creating-a-source-generator-part-8-solving-the-source-generator-marker-attribute-problem-part2/

**Comment by shahabganji** [comment]
If the analyzer depends on an external NuGet package for the generated source code, is there a possibility that somehow add that dependency to the NuGet package of the generator in order for the consumers of the generator to have that dependency automatically (in a transitive way) referenced? 

For instance, If the code generated by the generator depends on Newtonsoft.Json, then when a project adds the generator NuGet package, it automatically has the reference to the Newtonsoft.Json.

**Comment by MCGPPeters** [comment]
A cheap trick that worked for me is generating the attribute as a partial class... Not pretty... but it does the job...

**Reply by MichalBrylka** [reply]
How about emitting attribute to different namespace (like "ProjectRootNamespace".Attributes) - or maybe even control this namespace (or whether emitting should take place) via CompilerVisibleProperty ?

[a comment by an automated account is not shown]

**Comment by MichalBrylka** [comment]
How about emitting attribute to different namespace (like "ProjectRootNamespace".Attributes) - or maybe even control this namespace (or whether emitting should take place) via CompilerVisibleProperty ?

**Comment by elTRexx** [comment]
Hi. pardon me, but I don't find anywhere if you explain when and how you added Conditional to the attribute.
It appear the first time in the code on part 7 of this series, but without explanations on how you end up using it a first place and where you define the associated STRONGLY_TYPED_ID_USAGES value.
Because from what I read from your linked resources of yours, It seems to me a redundancy to the pre-compiler #if/#endif one, no ?

**Comment by AshbyGeek** [comment]
I just wrote a small source generator and took option 4. Worked well, thought I would comment on my experience a bit.

I wrote my source generator to not depend on the .dll that contained my attribute. Because of that, I was able to use msbuild to pack that dll into a nuget package as normal and just added one include to add the source generator's .dll into the nuget's `analyzers/dotnet/cs` folder. No funny business with privateAssets and ExcludeAssets.

**Reply by AshbyGeek** [reply]
Seriously though, thank you so much for the very detailed writeup on how you implemented your source generator and improved it along the way.
